x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypeConverter.java

```diff
@@ -382,10 +399,13 @@ public static Object convert(Object value, DataType dataType) {
         DATE_TO_DATETIME(value -> value),
 
         DATE_TO_BOOLEAN(delegate(DATETIME_TO_BOOLEAN)),
-        TIME_TO_BOOLEAN(fromTime(value -> value != 0));
+        TIME_TO_BOOLEAN(fromTime(value -> value != 0)),
+
+        STRING_ISO8601_PERIOD_TO_TEMPORAL(fromString(Period::parse, "duration")),
```

Comment:
Shouldn't "duration" be "interval" since this is used in the error message if the conversion is not possible?

Reply:
Also maybe use a static string variable.

Reply:
Did both, thanks.
(I've just stayed consistent with the local style previously, that uses "hard-coded" strings.)